test/e2e/nodenetworkstate_test.go

```diff
@@ -0,0 +1,69 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	nmstatev1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1"
+)
+
+var _ = Describe("NodeNetworkState", func() {
+	Context("when desiredState is configured", func() {
+
+		var desiredState nmstatev1.State
+
+		JustBeforeEach(func() {
+			for _, node := range nodes {
+				updateDesiredState(namespace, node, desiredState)
+			}
+		})
+		JustAfterEach(func() {
```

Comment:
This feels weird to me. Could you help Petr and tell him why is that better than having just BeforeEach and AfterEach?

Reply:
I am going to replace it by helper functions I don't like either and I hate myself for it.

Reply:
:open_mouth: